rplidar.go

```diff
@@ -1,36 +1,55 @@
-package serial
+package rplidar
 
 import (
+	"bufio"
 	"context"
 	"errors"
 	"fmt"
+	"image"
+	"image/color"
 	"math"
-	"sort"
+	"os"
+	"strings"
 	"sync"
 	"time"
 
-	"go.viam.com/rplidar"
 	"go.viam.com/rplidar/gen"
+	"go.viam.com/utils"
 
 	"github.com/edaniels/golog"
-	"github.com/golang/geo/r2"
-	"go.viam.com/core/config"
-	"go.viam.com/core/lidar"
-	"go.viam.com/core/registry"
-	"go.viam.com/core/robot"
-	"go.viam.com/core/usb"
+	"github.com/golang/geo/r3"
+	"go.viam.com/rdk/component/camera"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/robot"
+	"go.viam.com/rdk/spatialmath"
+	rdkUtils "go.viam.com/rdk/utils"
+)
+
+const (
+	// ModelName is how the lidar will be registered into core.
+	ModelName = "rplidar"
+
+	// Type is the lidar specific type.
+	Type = camera.SubtypeName
 )
 
 func init() {
-	registry.RegisterLidar(rplidar.ModelName, func(ctx context.Context, r robot.Robot, config config.Component, logger golog.Logger) (lidar.Lidar, error) {
-		return NewDevice(config.Host)
-	})
-	lidar.RegisterType(rplidar.Type, lidar.TypeRegistration{
-		USBInfo: &usb.Identifier{
```

Comment:
USBInfo (used in search device?) was performed here but removed in update

Reply:
Does that mean it has to be removed in general?

Reply:
nvm - put it outside of the init function and am using it for the search for the device.